Wangscape/noise/module/ReseedableOps.cpp

```diff
@@ -0,0 +1,502 @@
+        {"source", left},
+        {DEFAULT_MODULE_GROUP_OUT, scale_bias_p}
+    });
+}
+
+ReseedablePtr operator+(double left, ReseedablePtr right)
+{
+    return right + left;
+}
+
+ReseedablePtr& operator+=(ReseedablePtr& left, ReseedablePtr right)
+{
+    return (left = left + right);
+}
+
+ReseedablePtr& operator+=(ReseedablePtr& left, double right)
+{
+    return (left = left + right);
+}
+
+ReseedablePtr operator-(ReseedablePtr left, ReseedablePtr right)
+{
+    return left + -right;
+}
+
+ReseedablePtr operator-(ReseedablePtr left, double right)
+{
+    return left + -right;
+}
+
+ReseedablePtr operator-(double left, ReseedablePtr right)
+{
+    return left + -right;
+}
+
+ReseedablePtr operator-(ReseedablePtr source)
+{
+    return invert(source);
+}
+
+ReseedablePtr invert(ReseedablePtr source)
+{
+    auto invert_p = std::make_shared<Reseedable<Invert>>();
+    invert_p->module.SetSourceModule(0, source->getModule());
+
+    return makeModuleGroup({
+        {"source", source},
+        {DEFAULT_MODULE_GROUP_OUT, invert_p}
+    });
+}
+
+ReseedablePtr& operator-=(ReseedablePtr& left, ReseedablePtr right)
+{
+    return (left = left + -right);
+}
+
+ReseedablePtr& operator-=(ReseedablePtr& left, double right)
+{
+    return (left = left + -right);
+}
+
+ReseedablePtr operator*(ReseedablePtr left, ReseedablePtr right)
+{
+    auto multiply_p = std::make_shared<Reseedable<Multiply>>();
+    multiply_p->module.SetSourceModule(0, left->getModule());
+    multiply_p->module.SetSourceModule(1, right->getModule());
+
+    return makeModuleGroup({
+        {"source0", left},
+        {"source1", right},
+        {DEFAULT_MODULE_GROUP_OUT, multiply_p}
+    });
+}
+
+ReseedablePtr operator*(ReseedablePtr left, double right)
+{
+    auto scale_bias_p = std::make_shared<Reseedable<ScaleBias>>();
+    scale_bias_p->module.SetSourceModule(0, left->getModule());
+    scale_bias_p->module.SetScale(right);
+
+    return makeModuleGroup({
+        {"source", left},
+        {DEFAULT_MODULE_GROUP_OUT, scale_bias_p}
+    });
+}
+
+ReseedablePtr operator*(double left, ReseedablePtr right)
+{
+    return right * left;
+}
+
+ReseedablePtr& operator*=(ReseedablePtr& left, ReseedablePtr right)
+{
+    return (left = left * right);
+}
+
+ReseedablePtr& operator*=(ReseedablePtr& left, double right)
+{
+    return (left = left * right);
+}
+
+ReseedablePtr operator/(ReseedablePtr left, ReseedablePtr right)
+{
+    auto multiply_p = std::make_shared<Reseedable<Multiply>>();
+    multiply_p->module.SetSourceModule(0, left->getModule());
+    multiply_p->module.SetSourceModule(1, right->getModule());
+
+    return makeModuleGroup({
+        {"source0", left},
+        {"source1", right},
+        {DEFAULT_MODULE_GROUP_OUT, multiply_p}
+    });
+}
+
+ReseedablePtr operator/(ReseedablePtr left, double right)
+{
+    return (left * (1. / right));
+}
+
+ReseedablePtr operator/(double left, ReseedablePtr right)
+{
+    throw std::runtime_error("Not implemented");
```

Comment:
Static assert would probably be better, since this is never called. This was intended to be temporary, but now even if `Divide` modules are implemented this operator will be deleted.